### PR TITLE
Support passing tvdbid param via api, and retrieving info from tvmaze

### PR DESCRIPTION
**Pull request by miigotu**
Allows tvsearch with tvdbid, like other NN+ indexers.

### CODE REVIEW

**Comment by miigotu** [comment]
Ugh, let me fix tabs quick -.-

**Comment by miigotu** [comment]
will rebase this later. Just curious if there is interest in adding this?

**Comment by aroundmyroom** [comment]
Depends. Never use the search for tv series or it must be S01E01 .. 
I think sonarr does the job very well. 
I have sonarr, Jackett, nzEDB, deluge and some private trackers (and usenet) and that will find all tv shows I need .. 

But .. if it will help finding shows through sonarr .. Yup .. than I would be interested..

**Comment by miigotu** [comment]
I do see a small typo in the php here, will fix that.

however, tvsearch is designed to take an Id (used to be rageid as the rid param), a season, and an ep param.

newznab+ and nzbed indexers now support tvdbid search param now, so this PR is to reduce overall indexer load and increase accuracy of results as well as having a compatible set of parameters. a q param search is much less accurate.

sonarr is far from the only software making use of nzb indexer api's. You should also think of couchpotato, sickbeard, sickrage, sickgear, headphones, and countless other apps

**Comment by mesa57** [comment]
In my previous commit already TVRage id (rid) via TVMaze is supported.
For the time being applications can use TVRage id again (for instance Sickbeard).
It is still not clear to me if TVRage will re-appear.

**Comment by miigotu** [comment]
Im pretty confidant in confirming it will not return. Either way, I just think it makes sense to support tvdbid param using tvmaze as the mapper also since it is so simple to do.

I assumed the goal was to mimick the newznab api.
https://github.com/nZEDb/nZEDb/blob/c2a9cb42199846adacbbea83bdce4a95e0766f34/nzedb/Capabilities.php#L66
https://github.com/DariusIII/newznab-tmux/blob/ca6a7df390e412672db48ea3d1c2060c35fe6442/newznab/Capabilities.php#L66

File: lib/services/MediaInformation/Services_MediaInformation_Tvmaze.php [hunk 1]
```diff
@@ -3,24 +3,24 @@
...
+        * Create URL to retrive info from, for this provider
+        * we only support direct id lookups for now
+        */
         if ($this->getSearchName() == "tvmaze") {
-	        $url = 'http://api.tvmaze.com/shows/' . $this->getSearchid();
-        } else { 
-    	    $url = 'http://api.tvmaze.com/lookup/shows?tvrage=' . $this->getSearchid();
+            $url = 'http://api.tvmaze.com/shows/' . $this->getSearchid();
+        } else {
+            $url = 'http://api.tvmaze.com/lookup/shows?' . '=' $this->getSearchName() . $this->getSearchid();
```
Comment (miigotu) [inline lib/services/MediaInformation/Services_MediaInformation_Tvmaze.php]: aside from the rebase I need to do (shitty whitespace conflicts zzz), this line is supposed to be:
`$url = 'http://api.tvmaze.com/lookup/shows?' .  $this->getSearchName() . '='  . $this->getSearchid();`

**Comment by mesa57** [comment]
TVRage has returned, but their api is not yet operational yet.
If spotweb should support TVDbId searches, it should be via the TVDBId api and not via TVMaze.
Ultimately TVRage will be done via their api too.

**Comment by miigotu** [comment]
except its likely that most major indexers wont re-implement tvrage again...

**Comment by mesa57** [comment]
Could you resolve the conflicts with the main branch first ?

**Comment by miigotu** [comment]
Yeah, this is a pretty old PR now =P. I will see about updating it within the next day or do if it's wanted.

**Comment by mesa57** [comment]
If you want to, we can also aply for a public TVDB api key. 
Condition is that we ask users of spotweb to contribute to TVDB.

**Comment by miigotu** [comment]
I cant speak for the users of spotweb. TBH I dont even use it myself (or anything NZB related). I just did this because SickRage users asked me to =P

Will try and finish up my other work to update this PR soon

**Comment by mesa57** [comment]
It only means that if we want to use the TVDB api, we have to make a remark on for instance the release page (which anyway is outdated).

**Comment by mesa57** [comment]
As long as there are conflicts, I cannot merge this.
Are you willing to solve these conflicts ?
There is another pull request #179 which is waiting for that.
If you give no reaction in a few days I have to close this pull request.

**Comment by miigotu** [comment]
Sorry I dont have time to work on this. I had tried a while back and the conflict is pretty nasty now.

**Pull request closed without merging**